api/v1alpha1/httproute_types.go

```diff
@@ -115,15 +116,21 @@ type HTTPRouteMatch struct {
 	// Default: "/"
 	Path *string `json:"path" protobuf:"bytes,2,opt,name=path"`
 
-	// HeaderType defines the semantics of the `Header` matcher.
+	// HeaderMatchType defines the semantics of the `Header` matcher.
+	//
+	// Support: core (Exact)
```

Comment:
It would be useful to document the semantics with respect to whitespace and folding.  For example, "Neither leading nor trailing whitespace is stripped from the received header before comparison, and the comparison is performed in a case-sensitive manner (i.e., without case folding or normalization)."

Reply:
@jpeach commented (moving his comment to this thread):

> I don’t think that this is standards conforming. HTTP/1.1 has specific rules about white space in headers.

From Section 4.2 of RFC 2616:

> HTTP header fields, which include general-header (section 4.5),
   request-header (section 5.3), response-header (section 6.2), and
   entity-header (section 7.1) fields, follow the same generic format as
   that given in Section 3.1 of RFC 822 [9]. Each header field consists
   of a name followed by a colon (":") and the field value. Field names
   are case-insensitive. The field value MAY be preceded by any amount
   of LWS, though a single SP is preferred.

Not stripping anything from the value of the header makes sense to me and that is the norm that most proxies follow anyways.

Reply:
This is what I've put in currently, which is based on the above section from HTTP/1.1 spec:

```patch
+       // HeaderMatchTypeExact matches HTTP request-header fields.
+       // Neither leading or trailing whitespace is stripped.
+       // Field names matches are case-insensitive while field values matches
+       // are case-sensitive.
```

Reply:
Standards battle!

https://tools.ietf.org/html/rfc7230#section-3.2.4

>    A field value might be preceded and/or followed by optional
   whitespace (OWS); a single SP preceding the field-value is preferred
   for consistent readability by humans.  The field value does not
   include any leading or trailing whitespace: OWS occurring before the
   first non-whitespace octet of the field value or after the last
   non-whitespace octet of the field value ought to be excluded by
   parsers when extracting the field value from a header field.

Reply:
Let's see if the actual implementations will be good enough  in the initial pass to match RFC language wrt to matching...